decimal docs: specification link and examples (#128698)
python/cpython: Doc/library/decimal.rst

decimal floating-point arithmetic. It offers several advantages over the
:class:`float` datatype:

> * Decimal "is based on a [floating-point
>   model](https://speleotrove.com/decimal/damodel.html#refnumber) which was

Reviewer: In contrast, I think this link is more readable. Even though they are the same, I think this seems better from the perspective of the original text.
```suggestion
* Decimal "is based on a [floating-point model]
  (https://speleotrove.com/decimal/damodel.html#refnumber) which was
```

>   <https://speleotrove.com/decimal/damodel.html#refnumber>`_ which was designed

Reviewer 2: Prefer anonymous links:
```suggestion
  <https://speleotrove.com/decimal/damodel.html#refnumber>`__ which was designed
```

   >>> c % a
   Decimal('0.77')

> Decimals could be formatted in fixed-point or scientific notation, using same

Reviewer 3: ```suggestion
Decimals can be formatted in fixed-point or scientific notation, using the same
```
Change tense to present

> Decimals could be formatted in fixed-point or scientific notation, using same

Reviewer 3: ```suggestion
Decimals could be formatted in fixed-point or scientific notation, using the same
```
Grammatical error

formatting syntax (see :ref:`formatspec`) as builtin :class:`float` type:

.. doctest::

   >>> format(Decimal('2.675'), "f")
   '2.675'
   >>> format(Decimal('2.675'), ".2f")
   '2.68'
   >>> format(Decimal('2.675'), ".2e")
   '2.68e+0'
   >>> with localcontext() as ctx:
   ...     ctx.rounding = ROUND_DOWN
   ...     print(format(Decimal('2.675'), ".2f"))

>    ...

Reviewer 3: ```suggestion
```

Do we include the empty line? In this file there are 2 cases without it and 1 with it. Would it not be better to have consistency? 

```
>>> def mul(x, y, fp=TWOPLACES):
    ...     return (x * y).quantize(fp)
    ...

>>> def div(x, y, fp=TWOPLACES):
    ...     return (x / y).quantize(fp)

>>> def remove_exponent(d):
    ...     return d.quantize(Decimal(1)) if d == d.to_integral() else d.normalize()
```

Author: I'm not sure what will be consistent here.  This is just a copy of console session, I think it's fine.

>    ...

Reviewer 4: I think it would be worthwhile to mention f-strings and `str.format()` here. I don't think many people are familiar with `format`, and don't realize that all 3 use `obj.__format__` behind the scenes.
